railties/test/application/generators_test.rb

```diff
@@ -172,6 +172,14 @@ def with_bare_config
       assert File.exist?(File.join(rails_root, "app/views/notifier_mailer/foo.html.erb"))
     end
 
+    test "scaffold_controller generator with namespace and model" do
+      rails("generate", "model", "post", "title:string")
+      rails("db:migrate")
+      rails("generate", "scaffold_controller", "Admin::Post", "title:string", "--model-name=Post")
+      output = rails("test", "test/controllers/admin/posts_controller_test.rb")
+      assert_match(/0 failures, 0 errors/, output)
+    end
```

Comment:
Could we add some assertions here that demonstrate this fixes #43678?

Reply:
Hi @jonathanhefner, thanks for your feedback; I have already updated the code according to your suggestions.

What assertions are you thinking exactly? Checking that the partials `show` and `edit` contains the code `admin_post_path(@post)` and `edit_admin_post_path(@post)` for example? 

I can't imagine better tests than running the generated tests by the generator, but I am open to suggestions 😄 

It would be great to review this PR https://github.com/rails/jbuilder/pull/512. Both are pretty related and necessary to completely fix the generators when using namespaces.

Reply:
> What assertions are you thinking exactly? Checking that the partials `show` and `edit` contains the code `admin_post_path(@post)` and `edit_admin_post_path(@post)` for example?
>
> I can't imagine better tests than running the generated tests by the generator, but I am open to suggestions :smile:

Yes, sorry, I was automatically looking for `assert_file` calls and missed the `rails("test")` call.  There actually is a [`test_model_name_option`](https://github.com/rails/rails/blob/85f45350e6d3fb6284c744d36b77b79ce639dfee/railties/test/generators/scaffold_controller_generator_test.rb#L207) in `railties/test/generators/scaffold_controller_generator_test.rb`, but it should be extended to check additional files (e.g. `functional_test.rb`, `edit.html.erb`, `show.html.erb` `partial.html.erb`) to prevent such regressions.  

There are a [few places in `scaffold_controller_generator_test.rb`](https://github.com/rails/rails/blob/e8c0293a77dfe2c03bb32bfbd1dd01b2eb931a77/railties/test/generators/scaffold_controller_generator_test.rb#L236-L258) where we check the output of `bin/rails test`, but for different reasons.  I think we may prefer to check file contents rather than run `bin/rails test` due to the impact on test suite time.  Though, I agree, running `bin/rails test` might be more fool-proof (as long as you're sure the tests were generated in the first place!).

Reply:
I added a few assertions to `railties/test/generators/scaffold_controller_generator_test.rb`, let me know if you miss something.